### PR TITLE
Add an unsafe method view_mut_raw

### PR DESCRIPTION
**Pull request by lshlyapnikov**
See #1643
### TODO:
- [x] add `Uint8Array::view_mut_raw` test
- [x] add `Int16Array::view_mut_raw` test
- [x] add `Uint32Array::view_mut_raw` test
- [x] add `Float64Array::view_mut_raw` test

### CODE REVIEW

File: crates/js-sys/tests/wasm/Array.rs [hunk 1]
```diff
@@ -466,3 +466,8 @@ fn array_inheritance() {
     assert!(array.is_instance_of::<Object>());
     let _: &Object = array.as_ref();
 }
+
+#[wasm_bindgen_test]
+fn uint8_view_mut_raw() {
+    assert!(1 == 2);
```
Comment (lshlyapnikov) [inline crates/js-sys/tests/wasm/Array.rs]: @alexcrichton How do I run the tests locally?
`cargo test -p js-sys` does not run anything:
```
$ cargo test -p js-sys
    Finished dev [unoptimized + debuginfo] target(s) in 0.06s
     Running target/debug/deps/headless-cb15791b0fc6e9fb

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out

     Running target/debug/deps/wasm-d7d4f9b8001c0920

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out
```
```
$ cargo +nightly test --target wasm32-unknown-unknown -p js-sys
   Compiling cfg-if v0.1.10
   Compiling scoped-tls v1.0.0
   Compiling wasm-bindgen-backend v0.2.53 (/home/lshlyapnikov/Projects/rustwasm/wasm-bindgen/crates/backend)
error[E0463]: can't find crate for `core`
  |
  = note: the `wasm32-unknown-unknown` target may not be installed

error: aborting due to previous error

For more information about this error, try `rustc --explain E0463`.
error: could not compile `cfg-if`.
warning: build failed, waiting for other jobs to finish...
error[E0463]: can't find crate for `std`
  |
  = note: the `wasm32-unknown-unknown` target may not be installed

error: aborting due to previous error

For more information about this error, try `rustc --explain E0463`.
error: could not compile `scoped-tls`.
warning: build failed, waiting for other jobs to finish...
error: build failed
```

File: crates/js-sys/src/lib.rs [hunk 2]
```diff
@@ -4792,6 +4792,16 @@ macro_rules! arrays {
                 )
             }
 
+            pub unsafe fn view_mut_raw(ptr: *mut $ty, length: usize) -> $name {
```
Comment (lshlyapnikov) [inline crates/js-sys/src/lib.rs]: @alexcrichton do we need it for all arrays: `Int8Array`, `Int16Array`,... `Uint8Array` or just for `Uint8Array`?

File: crates/js-sys/tests/wasm/Array.rs (hunk 1 above)
Comment (alexcrichton) [reply crates/js-sys/tests/wasm/Array.rs]: You'll want to `rustup target add wasm32-unknown-unknown` I believe

File: crates/js-sys/src/lib.rs (hunk 2 above)
Comment (alexcrichton) [reply crates/js-sys/src/lib.rs]: That's up to you! I think it's fine to add it to all though

**Comment by alexcrichton** [comment]
Looking good!

I'd be happy to merge when you're ready!

File: crates/js-sys/src/lib.rs [hunk 3]
```diff
@@ -4792,6 +4792,31 @@ macro_rules! arrays {
...
+            ///
+            /// # Unsafety
+            ///
+            /// Views into WebAssembly memory are only valid so long as the
+            /// backing buffer isn't resized in JS. Once this function is called
+            /// any future calls to `Box::new` (or malloc of any form) may cause
+            /// the returned value here to be invalidated. Use with caution!
+            ///
+            /// Additionally the returned object can be safely mutated,
+            /// the changes are guranteed to be reflected in the input array.
```
Comment (lshlyapnikov) [inline crates/js-sys/src/lib.rs]: @alexcrichton please doublecheck the rustdoc.

File: crates/js-sys/src/lib.rs [hunk 4]
```diff
@@ -4792,6 +4792,31 @@ macro_rules! arrays {
...
+            pub unsafe fn view_mut_raw(ptr: *mut $ty, length: usize) -> $name {
```
Comment (lshlyapnikov) [inline crates/js-sys/src/lib.rs]: @alexcrichton please doublecheck the rustdoc.

**Comment by alexcrichton** [comment]
Looks great to me, thanks!

**Comment by mstange** [comment]
Thank you!

**Pull request merged**